### DISCUSSION TITLE
Custom script - remove primary IP on an VM or Device?

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#14412] [Help Wanted!] [answered]
Simple thing really, but I'm lost. I have a need to remove current primary ip, and set another using an custom script. But I can't figure out how. 

Example, getting an VirtualMachine object and trying to do "x.primary_ip = None" gives an "AttributeError: can't set attribute 'primary_ip'"

**Comment by diniramdhni** [comment]
```
# Assuming you have a VM object
virtual_machine = get_virtual_machine()

# Check if the virtual machine has a method to change the primary IP
if hasattr(virtual_machine, 'set_primary_ip'):
    # Call the method to set a new primary IP
    virtual_machine.set_primary_ip(new_primary_ip)
    print("Primary IP updated successfully.")
else:
    print("The virtual machine does not support updating the primary IP directly.")
```
Replace `get_virtual_machine()` with the method you actually use to get the VM object, and `set_primary_ip` with the function given by your virtualization library or API. If no such method exists, you may need to study the documentation for the library or API you're using, as well as the documentation for your virtualization platform.

**Reply by ziggekatten** [reply]
well. I try do do this with Netbox Custom script / nbshell, and I can't find any functions for setting / removing primary IP on the object

**Comment by ziggekatten** [comment, marked as answer]
And I'm so stupid...... object.primary_ip is read only..... object.primary_ip4 and object.primary_ip6 is not and only have to be set to None....

Well...sorted.
